django/db/models/window.py

```diff
@@ -0,0 +1,190 @@
+from django.db.models.expressions import (
+    BaseExpression, Expression, ExpressionList, OrderBy,
+)
+from django.utils.functional import cached_property
+
+__all__ = ['RowRange', 'ValueRange', 'Window']
+
+
+class Window(Expression):
+    template = '%(expression)s OVER (%(window)s)'
+    filterable = False
+
+    def __init__(self, expression, partition_by=None, order_by=None, frame=None, output_field=None):
+        self.partition_by = partition_by
+        self.order_by = order_by
+        self.frame = frame
+
+        if not getattr(expression, 'window_compatible', False):
+            raise ValueError('expression must be compatible in an over-clause')
+
+        if self.partition_by is not None:
+            if isinstance(self.partition_by, (tuple, list)):
+                self.partition_by = ExpressionList(*self.partition_by)
+            else:
+                self.partition_by = ExpressionList(self.partition_by)
+
+        if self.order_by is not None:
+            if not isinstance(self.order_by, (BaseExpression, OrderBy, list, tuple)):
```

Comment:
You can remove `OrderBy` from here because it's a subclass of `BaseExpression`. Also could use `collections.Sequence` instead of `list` and `tuple` to allow anything that the * operator will work with.

The message could also read 'must be either an Expression or a sequence of expresssions'